libs/dotnet-sdk/UpdateStrategies/OnRequestUpdateStrategyMiddleware.cs

```diff
@@ -6,13 +6,13 @@
 
 namespace FeatureBoard.DotnetSdk.UpdateStrategies;
 
-public class OnRequestUpdateStrategyMiddleware
+public class OnRequestUpdateStrategyMiddleware : IMiddleware
```

Comment:
This makes it per client request and no longer a singleton (is this intended)
https://learn.microsoft.com/en-us/aspnet/core/fundamentals/middleware/extensibility?view=aspnetcore-7.0

Reply:
Yep, that is the intention. `HttpClient` has an only somwhat resolved bug (feature?) whereby [DNS changes are never discovered if `HttpClient` is used as a singleton](https://github.com/dotnet/runtime/issues/18348) (this would be a concern if FeatureBoard API is using blue/green deployments for example). There are two solutions for this
1. [Configure the underlying `SocketsHttpHandler.PooledConnectionLifetime`](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-3.1#alternatives-to-ihttpclientfactory-2) or 
2. Make `HttpClient` have lifetime short

Here I've elected the latter since it also plays more nicely with concept of having distinct named/typed clients (see above comment for concerns with using shared ones) and the Polly based retry extensibility points. Given this middleware depends on `FeatureBoardHttpClient` which in turn depends on `HttpClient`, everything in this dependency chain must be scoped (if anything in the chain becomes singleton the things further down the dependency graph will become ["captured"](https://blog.ploeh.dk/2014/06/02/captive-dependency/).

Reply:
👍